OneCommunityGlobal/HighestGoodNetworkApp
# Nathan reset permissions to default

# Description
This PR addresses `Create a “Restore To Default” functionality that restores the user permissions to the default configurations`.

## Related PRS (if any):
This frontend PR is related to backend PR #559.
To test this frontend PR you need to checkout the backend PR.
https://github.com/OneCommunityGlobal/HGNRest/pull/559

## Main changes explained:
- Created `PermissionsPresetsModal.jsx` for updating permissions of roles to various presets
- Added redux logic for storing the presets of the role the user last viewed
- Added `Load Presets` and `Create New Preset` buttons in `RolePermissions.jsx`
- `Load Presets` opens the `PermissionsPresetsModal` and `Create New Preset` creates a new preset with the current permissions and auto assigns it the name of `New Preset #` with the lowest number that isn't taken.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner user
4. go to dashboard→ Other Links→ Permissions Management→any role, preferably a test role (I've been using DNE)
5. Verify each function (and that their changes persist after refresh if saved)
    1. `Create New Preset` and `Load Presets` buttons appear next to the `Save` and `Delete Role` buttons.
    2. `Load Presets` opens the `PermissionsPresetsModal` and loads presets for that role from the back end.
    3. `Create New Preset` saves the current permissions in state (regardless if they are not saved to be applied to the actual role) as a new preset which is visible when you open the `PermissionsPresetsModal`
    4. Verify `PermissionsPresetsModal` works
        1. A list of permissions will expand and collapse when the preset is clicked (on an area not covered by a button or input field)
        2. When the edit button is clicked, it will toggle the following features: the name for that preset becomes an input field, is selected, and has a `Save Name` button appear
        3. clicking `Save Name` will save name changes after refresh
        4. The `Delete` button removes presets
        5. The `Apply` button applies presets 


## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/71ea27de-aba9-41e9-8200-54fd82955b9b
